server/server.go

```diff
@@ -336,12 +307,21 @@ func (s *Server) Run() error {
 	if s.cfg.Status.ReportStatus {
 		s.startStatusHTTP()
 	}
+	go s.startNetworkListener(s.listener)
+	go s.startNetworkListener(s.socket)
```

Comment:
If `s.cfg.Socket` is not configured, will `s.socket` be a nil pointer?

Reply:
I think so, let me verify and come back.

Reply:
> If `s.cfg.Socket` is not configured, will `s.socket` be a nil pointer?

Yes it is, check both through debugger as well as logic (just start tidb-server without arguments and it will not try to open any socket file and just return from `startNetworkListener(s.socket)`